### PR TITLE
Csi 884

### PR DESCRIPTION
**Pull request by HarishRocks**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-884 The coin value doesn't re-sync up automatically even after 15 min.](https://app.clickup.com/t/37308523/CSI-884)

File: cysync/src/store/database/helper.ts [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
 
-  return priceDb.getPrice(coin.toLowerCase(), 7).then(res => {
-    if (res) {
-      const { length } = res.data;
-      return res.data[length - 1][1];
-    }
-    logger.warn(`Cannot find price for coin ${coin}`);
-    return 0;
-  });
+  return (await latestPriceDb.getPrice(coin)).price;
```
Comment (irshadCypherock) [inline cysync/src/store/database/helper.ts:8]: Keep the previous checks as it is.

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: cysync/src/store/database/helper.ts (hunk 1 above)
Comment (HarishRocks) [reply cysync/src/store/database/helper.ts]: Got it. Pls check the latest commit @irshadCypherock

**Review by irshadCypherock — CHANGES_REQUESTED** [review]
Latest prices are not fetched when a coin is added.

**Comment by HarishRocks** [comment]
Hey @irshadCypherock can you check the latest commit

File: cysync/src/mainProcess/utils.ts [hunk 2]
```diff
@@ -128,16 +129,21 @@ export const fadeInWindow = (
 ) => {
   let currentOpacity = window.getOpacity();
 
-  const timerID = setInterval(() => {
-    if (!window || window.isDestroyed()) {
-      clearInterval(timerID);
-      return;
-    }
+  const timerID = useRef(null);
```
Comment (irshadCypherock) [inline cysync/src/mainProcess/utils.ts:132]: Don't use ref in main process.

File: cysync/src/store/provider/syncProvider/index.tsx [hunk 3]
```diff
@@ -998,46 +998,48 @@ export const SyncProvider: React.FC = ({ children }) => {
     await notifications.getLatest();
   };
 
-  const intervals: NodeJS.Timeout[] = [];
+  const emptyIntervals: NodeJS.Timeout[] = [];
```
Comment (irshadCypherock) [inline cysync/src/store/provider/syncProvider/index.tsx:1001]: Don't store the reference to empty intervals

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: cysync/src/mainProcess/utils.ts (hunk 2 above)
Comment (HarishRocks) [reply cysync/src/mainProcess/utils.ts]: Okay

File: cysync/src/store/provider/syncProvider/index.tsx (hunk 3 above)
Comment (HarishRocks) [reply cysync/src/store/provider/syncProvider/index.tsx]: Yeah cool

File: cysync/src/store/provider/syncProvider/index.tsx [hunk 4]
```diff
@@ -929,34 +994,51 @@ export const SyncProvider: React.FC = ({ children }) => {
     await addBalanceRefresh({ isRefresh: true });
     await addHistoryRefresh({ isRefresh: true });
     await addPriceRefresh({ isRefresh: true });
+    await addLatestPriceRefresh({ isRefresh: true });
     await notifications.getLatest();
   };
 
-  let interval: NodeJS.Timeout | undefined;
+  const intervals = useRef([] as NodeJS.Timeout[]);
```
Comment (irshadCypherock) [inline cysync/src/store/provider/syncProvider/index.tsx:1001]: To use types here do. `useRef<NodeJS.Timeout[]>([])`

**Pull request merged**